packages/discord-utilities/src/lib/CommandInteractionOptionResolver.ts

```diff
@@ -0,0 +1,381 @@
+import {
+	ApplicationCommandOptionType,
+	ApplicationCommandType,
+	InteractionType,
+	type APIApplicationCommandAutocompleteInteraction,
+	type APIApplicationCommandInteraction,
+	type APIApplicationCommandInteractionDataBasicOption,
+	type APIApplicationCommandInteractionDataIntegerOption,
+	type APIApplicationCommandInteractionDataNumberOption,
+	type APIApplicationCommandInteractionDataOption,
+	type APIApplicationCommandInteractionDataStringOption,
+	type APIAttachment,
+	type APIChatInputApplicationCommandInteractionDataResolved,
+	type APIInteractionDataResolvedChannel,
+	type APIInteractionDataResolvedGuildMember,
+	type APIMessage,
+	type APIMessageApplicationCommandInteractionDataResolved,
+	type APIModalSubmitInteraction,
+	type APIRole,
+	type APIUser,
+	type APIUserApplicationCommandInteractionDataResolved
+} from 'discord-api-types/v10';
+
+/**
+ * Utility class for resolving command interaction options while working with the raw API.
+ * Based on {@linkplain https://github.com/discordjs/discord.js/blob/main/packages/discord.js/src/structures/CommandInteractionOptionResolver.js}
+ */
+export class CommandInteractionOptionResolver {
```

Comment:
The naming isn't really accurate.. and why is ModalSubmit here too, but message component ones aren't?

Reply:
> but message component ones aren't?

What would there be? I suppose with the new select menus there's something, but other than that there isn't much to parse out of message components.

With that said, what naming are you in favor of?

Reply:
well modal submits are not part of command interactions 😓 
Either split them up, or just call this InteractionOptionresolver? @favna might have better ideas too

Reply:
I'd say indeed either split it off or rename it. Don't really know how to name it other than what @vladfrangu suggested.

Reply:
Renamed.